### PR TITLE
Use webpack to improve performance

### PR DESCRIPTION
**Pull request by Flanker32**
Use webpack to improve performance

### CODE REVIEW

**Comment by yaohaizh** [comment]
@jdneo, @Eskibear please help review this PR.

File: package.json [hunk 1]
```diff
@@ -108,11 +108,14 @@
...
   "scripts": {
-    "vscode:prepublish": "tsc -p ./",
     "compile": "tsc -watch -p ./",
+    "watch": "webpack --mode development --watch --info-verbosity verbose",
     "postinstall": "node ./node_modules/vscode/bin/install",
     "test": "node ./scripts/download-vscode-for-system-tests && node ./scripts/install-vsix-dependencies redhat.java && node ./scripts/install-vsix-dependencies vscode-java-dependency-0.2.0.vsix && node ./scripts/run-vscode-tests",
-    "build-server": "./node_modules/.bin/gulp build_server"
+    "build-server": "./node_modules/.bin/gulp build_server",
+    "vscode:prepublish": "tsc -p ./ && webpack --mode production",
+    "webpack": "webpack --mode development",
```
Comment (jdneo) [inline package.json]: Are `webpack` and `webpack-dev` used anywhere?

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        filename: 'extension.js',
+        libraryTarget: "commonjs2",
+        devtoolModuleFilenameTemplate: "../[resource-path]",
+    },
+    externals: {
+        vscode: "commonjs vscode", // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+    },
+    devtool: 'source-map',
+    resolve: { // support reading TypeScript and JavaScript files, 📖 -> https://github.com/TypeStrong/ts-loader
+        extensions: ['.ts', '.js', '.json'],
```
Comment (jdneo) [inline webpack.config.js]: Do we have `*.json` files?

**Comment by yaohaizh** [comment]
test this please

**Comment by yaohaizh** [comment]
okay to test

**Pull request closed without merging**